sklearn/utils/_set_output.py

```diff
@@ -125,9 +126,10 @@ def _wrap_data_with_container(method, data_to_wrap, original_input, estimator):
         return data_to_wrap
 
     # dense_config == "pandas"
+    index = original_input.index if _is_pandas_df(original_input) else None
     return _wrap_in_pandas_container(
         data_to_wrap=data_to_wrap,
-        index=getattr(original_input, "index", None),
+        index=index,
```

Comment:
Hi @thomasjpfan,

I have had a regression since 1.3.1 and located that in those lines of code.
My understanding is the following but I could be mistaken so please correct me: those lines have changed the set_output behaviour such that
- before, if original_input had an index attribute (as is unfortunately the case for lists, but also for pd.Series), the index gets wrapped, and bad things ensued for lists,
- now, if original_input is not a pd.DataFrame (including intented targets lists but unfortunately also pd.Series), the index does not get wrapped.

So while the fix is valuable for lists, it is not for the case where input are pd.Series.
Would you agree? and if so is a good way to solve this changing from `is_pandas_df` to a such function `is_pandas_df_or_series`?

Thank you for your attention.

Reply:
The `set_output` API was only designed with 2-D containers in mind. This means it does not consider `pd.Series`.

If you have an use case for `set_output` with a 1-D container, may you open an issue about it?

Reply:
Oh I did not know that, OK thank you for your feedback!